### PR TITLE
[ios, macos] Fix MGLLight.anchor to accept style functions

### PR DESCRIPTION
**Pull request by fabian-guerra**
Fixes https://github.com/mapbox/mapbox-gl-native/issues/9085

### CODE REVIEW

**Comment by jfirebaugh** [comment]
This is intended to be targeted at the release branch, right?

**Comment by fabian-guerra** [comment]
@jfirebaugh yesterday I pushed this branch thinking that it contained the fix I made, for an unknown reason my commit got lost. You are right this branch should be merged into `release-ios-v3.6.0-android-v5.1.0`

File: platform/darwin/test/MGLLightTest.mm [hunk 1]
```diff
@@ -25,7 +25,10 @@ - (void)testProperties {
     {
         mbgl::style::Light light;
         MGLLight *mglLight = [[MGLLight alloc] initWithMBGLLight:&light];
-        XCTAssertEqual(mglLight.anchor, MGLLightAnchorViewport);
+        
+        NSAssert([mglLight.anchor isKindOfClass:[MGLConstantStyleValue class]], @"mglLight.anchor isn’t a MGLConstantStyleValue.");
```
Comment (boundsj) [inline platform/darwin/test/MGLLightTest.mm]: Since this is a test, would it make more sense to use an XCT assertion instead of NSAssert?

Comment (1ec5) [inline platform/darwin/test/MGLLightTest.mm]: That’s a good point. A failed assertion would crash the test harness instead of outputting an error.

Comment (fabian-guerra) [inline platform/darwin/test/MGLLightTest.mm]: @boundsj @1ec5 sorry I didn't see this until recently. I will fix this in a separate branch.

**Pull request merged**